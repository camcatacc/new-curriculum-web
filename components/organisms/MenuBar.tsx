import React from "react";
import { secondary } from "../../styles/customColors";
import { PageSize } from "../atoms/PageSize";
import { ListMenuElements } from "../molecules/ListMenuElements";

export const MenuBar = () => {
    return (
        <div
            style={{ backgroundColor: secondary }}
            className="w-full sticky top-0 flex justify-center pt-3 pb-3 z-50"
        >
            <PageSize>
                <ListMenuElements menuElementsNames={["INICI", "QUI SÃ“C"]} />
            </PageSize>
        </div>
    );
};
