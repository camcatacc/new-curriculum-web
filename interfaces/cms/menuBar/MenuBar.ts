// Constants
export const CONTENT_TYPE_MENU_BAR = "menu";

// Definitions
import type { ContentfulEntry } from "interfaces/cms/contentful";

export interface CmsMenuBar {
	elements: ContentfulEntry<{ name: string; path: string }>[];
}
