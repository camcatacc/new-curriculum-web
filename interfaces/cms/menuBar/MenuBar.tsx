export const CONTENT_TYPE_MENU_BAR = "menu";
export interface CmsMenuBar {
	elements: { fields: { name: string; path: string } }[];
}
