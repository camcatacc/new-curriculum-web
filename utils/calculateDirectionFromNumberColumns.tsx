// TODO: Simplify this function
function calculateDirectionFromNumberColumns(i: number, total: number, numPerRow: number): "fromLeft" | "fromRight" | "fromBottom" {
	// If everything does not fit in a row
	if (numPerRow < total) {
		// We calculate the number of items in the last row
		let nItemsLastRow = total % numPerRow;
		// We calculate the position from which the items will be in the last row
		let threshold = total - nItemsLastRow;
		// And we calculate the item position in that row
		let posInNewRow = i - numPerRow;

		// If the threshold is smaller than the item position and there are remaining items to be placed enter this conditional
		// If not, then the default method can be applied
		if (threshold < i && nItemsLastRow !== 0) {
			// If there is only one item in the last row, it must come from the bottom
			if (nItemsLastRow === 1) return "fromBottom";
			// We use the default method BUT with the position of the item IN THE LAST ROW
			switch (posInNewRow % nItemsLastRow) {
				case 0:
					return "fromRight";
				case 1:
					return "fromLeft";
				default:
					return "fromBottom";
			}
		}
	}

	switch (
		i % numPerRow //
	) {
		case 0:
			return "fromRight";
		case 1:
			return "fromLeft";
		default:
			return "fromBottom";
	}
}

export default calculateDirectionFromNumberColumns;
