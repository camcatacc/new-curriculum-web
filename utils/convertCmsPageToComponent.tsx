// Modules
import React from "react";

// Elements
import Abilities from "components/templates/Abilities/Abilities";
import Presentation from "components/templates/Presentation/Presentation";
import FindMe from "components/templates/FindMe/FindMe";

// Definitions
import type { ContentfulEntry } from "interfaces/cms/contentful";
import type { CmsPage } from "interfaces/cms/pages/Page";
import type { CmsPresentationPage } from "interfaces/cms/pages/Presentation";
import type { CmsAbilitiesPage } from "interfaces/cms/pages/Abilities";
import type { CmsFindMePage } from "interfaces/cms/pages/FindMe";

// Functions
const convertCmsPageToComponent = (page: ContentfulEntry<CmsPage>) => {
	switch (page.fields.type) {
		case "PresentationPage":
			return presentationCmsToComponent(page.fields.content.fields);
		case "AbilitiesPage":
			return abilitiesCmsToComponent(page.fields.content.fields);
		case "FindMePage":
			return findMeCmsToComponent(page.fields.content.fields);
	}
};

const presentationCmsToComponent = (page: CmsPresentationPage) => <Presentation {...page} />;
const abilitiesCmsToComponent = (page: CmsAbilitiesPage) => {
	const lists = page.abilitiesLists.map((abList) => {
		return {
			...abList.fields
		};
	});

	return <Abilities title={page.title} abilitiesLists={lists} />;
};

const findMeCmsToComponent = (page: CmsFindMePage) => <FindMe {...page} />;

export default convertCmsPageToComponent;
