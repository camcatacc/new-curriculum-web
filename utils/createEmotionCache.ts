// Modules
import createCache from "@emotion/cache";

// Functions
export default function createEmotionCache() {
	return createCache({ key: "css" });
}
