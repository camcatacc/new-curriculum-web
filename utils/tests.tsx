// Modules
import React, { ReactElement } from "react";

// Testing modules
import { render, RenderOptions, RenderResult } from "@testing-library/react";

// Redux Modules
import { configureStore } from "@reduxjs/toolkit";

// Redux Components
import rootReducer from "redux/rootReducer";
import { Provider } from "react-redux";
import { defaultTestRootState } from "redux/store.test";

// Definitions
import type { RootState } from "redux/types";

// Functions
export const customRender = (ui: ReactElement, state: Partial<RootState> = defaultTestRootState, renderOptions?: RenderOptions): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		const store = configureStore({
			reducer: rootReducer,
			preloadedState: state
		});

		return <Provider store={store}>{children}</Provider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
