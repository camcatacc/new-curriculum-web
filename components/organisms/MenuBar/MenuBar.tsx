// Modules
import React, { useRef } from "react";
import { useSelector } from "react-redux";
import { isNightModeSelector, switchNightMode } from "redux/ui/ui";
import { useAppDispatch } from "redux/store";

// Elements
import ListMenuElements from "components/molecules/ListMenuElements/ListMenuElements";

// Styles
import { secondary } from "styles/customColors";
import * as Styled from "components/organisms/MenuBar/MenuBar.styled";

// Definitions
export interface MenuBarProps {
	elements: { name: string; id: string; onClick: () => void }[];
	selectedId: string;
}

// Element
const MenuBar = ({ elements, selectedId }: MenuBarProps) => {
	const ref = useRef<HTMLDivElement>(null);
	const nightMode = useSelector(isNightModeSelector);
	const dispatch = useAppDispatch();

	const onSwitchChange = (ev: React.ChangeEvent<HTMLInputElement>) => dispatch(switchNightMode(ev.target.checked));

	return (
		<div ref={ref} style={{ backgroundColor: secondary, transition: "top 1s" }} className="w-full sticky top-0 flex justify-center pt-2 pb-2 z-50">
			<ListMenuElements menuElements={elements} selectedId={selectedId} className="w-4/5 max-w-screen-lg items-center" />
			<Styled.MaterialUISwitch checked={nightMode} onChange={onSwitchChange} value="nightMode" />
		</div>
	);
};

export default MenuBar;

// Hides the bar menu when scrolled down and shows it when scrolled up

/* 	useEffect(() => {
		var prevScrollPosition = window.scrollY;
		window.onscroll = () => {
			var currentScrollPos = window.scrollY;
			if (ref.current) ref.current.style.top = prevScrollPosition > currentScrollPos ? "0" : "-100px";
			prevScrollPosition = window.scrollY;
		};
	});
 */
