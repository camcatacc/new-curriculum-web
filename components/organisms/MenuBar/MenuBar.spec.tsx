// Modules
import React from "react";

// Testing modules
import { customRender } from "utils/tests";

// Elements
import MenuBar from "components/organisms/MenuBar/MenuBar";

// Definitions
import type { MenuBarProps } from "components/organisms/MenuBar/MenuBar";

// Default Props
const defaultProps: MenuBarProps = {
	elements: [
		{ name: "Home", onClick: jest.fn(), id: "home" },
		{ name: "About", onClick: jest.fn(), id: "about" }
	],
	selectedId: "home"
};

// Tests
describe("MenuBar", () => {
	beforeAll(() => {
		Object.defineProperty(window, "matchMedia", {
			writable: true,
			value: jest.fn().mockImplementation((query) => ({
				matches: false,
				media: query,
				onchange: null,
				addListener: jest.fn(), // Deprecated
				removeListener: jest.fn(), // Deprecated
				addEventListener: jest.fn(),
				removeEventListener: jest.fn(),
				dispatchEvent: jest.fn()
			}))
		});
	});
	it("Renders as expected if width > 720", () => {
		const { container } = customRender(<MenuBar {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
