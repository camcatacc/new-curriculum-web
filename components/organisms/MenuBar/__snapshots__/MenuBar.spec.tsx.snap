// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar Renders as expected if width > 720 1`] = `
<div>
  <div
    class="sc-gsDKAQ jWzHub w-full sticky top-0 flex justify-center pt-2 pb-2 z-50"
  >
    <div
      class="w-4/5 max-w-screen-lg items-center flex-row flex gap-28"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-bdvvtL cRyQUS cursor-pointer font-semibold uppercase MuiTypography-colorTextSecondary"
        >
          Home
        </a>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-bdvvtL nvKmn cursor-pointer font-regular uppercase MuiTypography-colorTextSecondary"
        >
          About
        </a>
      </p>
    </div>
    <span
      class="MuiSwitch-root WithStyles(ForwardRef(Switch))-root-1 WithStyles(ForwardRef(Switch))-root-2"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-6 MuiSwitch-input"
            type="checkbox"
            value="nightMode"
          />
          <span
            class="MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiSwitch-track"
      />
    </span>
  </div>
</div>
`;
